webui/react/src/hooks/useSettings.ts

```diff
@@ -181,18 +181,16 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
   useEffect(() => {
     if (!querySettings) return;
 
-    const settings = queryToSettings<T>(config, querySettings);
-    const stateSettings = state ?? {};
-
-    if (isEqual(settings, stateSettings)) return;
-
-    Object.keys(settings).forEach((setting) => {
-      stateSettings[setting] = settings[setting];
+    const parsedSettings = queryToSettings<T>(config, querySettings);
+    stateOb.update((s) => {
+      const oldSettings = s.get(config.storagePath) ?? {};
+      const newSettings = { ...s.get(config.storagePath), ...parsedSettings };
+      return s.set(
+        config.storagePath,
+        isEqual(oldSettings, newSettings) ? oldSettings : newSettings,
+      );
```

Comment:
Looks good to me, and this also becomes consistent with the usage at `updateSettings`

Reply:
✔️  defined this later in the file so it can use `updateSettings`